src/renderers/shared/reconciler/ReactCompositeComponent.js

```diff
@@ -278,6 +284,44 @@ var ReactCompositeComponentMixin = {
     this._pendingReplaceState = false;
     this._pendingForceUpdate = false;
 
+    var markup;
+    if (inst.handleError) {
+      var checkpoint = transaction.checkpoint();
+      try {
+        markup = this.performInitialMount(renderedElement, nativeParent, nativeContainerInfo, transaction, context);
```

Comment:
This only adds a single stack frame to the boundary, not every component so I think it is probably fine.